syntax = "proto2";

import "google/protobuf/descriptor.proto";

message FileResponseInfo
{
    /* The path to the file in the file system */
    optional string file_path = 1;
}

// This extension can be used to cause the autogenerated API functions to
// gather the message response from a file in the file system, rather than
// actually calling the RPC and receiving the data from a server via a socket.
// It is assumed (and required) that the file contains the expected response
// message type serialised via the 'cmsg_dump_msg_to_file' function. If the
// file does not exist then an empty message with ANT_CODE_OK set will be
// returned.
//
// Note that to use this extension the method must return a message with a
// nested 'ant_result' message.
extend google.protobuf.MethodOptions {
    optional FileResponseInfo file_response = 72295733;
}
